descartes_planner/src/planning_graph.cpp

```diff
@@ -36,16 +36,108 @@
 #include <boost/uuid/uuid_io.hpp> // streaming operators
 #include <boost/graph/dijkstra_shortest_paths.hpp>
 
+#include <future>
+
 using namespace descartes_core;
 using namespace descartes_trajectory;
+
+namespace
+{
+  using JointSolutions = std::vector<std::vector<double>>;
+  using VecJointSolutions = std::vector<JointSolutions>;
+  using ConstTrajectoryIterator = std::vector<descartes_core::TrajectoryPtPtr>::const_iterator;
+
+  using JointSolutionResult = std::pair<VecJointSolutions, bool>; // result and flag if everything went well
+
+  JointSolutionResult
+  rangeCalculateJointSolutions(descartes_core::RobotModelPtr model, 
+                               ConstTrajectoryIterator begin,
+                               ConstTrajectoryIterator end)
+  {
+    JointSolutionResult solutions;
+    solutions.first.reserve(std::distance(begin, end));
+    solutions.second = true;
+
+    while (begin != end)
+    {
+      JointSolutions joint_poses;
+      begin->get()->getJointPoses(*model, joint_poses);
+
+      if (joint_poses.empty())
+      {
+        solutions.second = false;
+        ROS_ERROR("No joint solutions available for point %s", 
+          boost::uuids::to_string(begin->get()->getID()).c_str());
```

Comment:
This isn't necessarily an issue but in order to avoid doing all these conversions into a c_string you could use the ROS_???_STREAM variant for printing uuids since the uuid_io header already provides an overloaded insertion operator that works with those.